service/src/main/java/com/github/swapbook/repositories/book/FakeBookRepository.java

```diff
@@ -0,0 +1,76 @@
+package com.github.swapbook.repositories.book;
+
+import com.github.swapbook.model.Book;
+import com.github.swapbook.model.Specimen;
+import com.github.swapbook.repositories.specimen.SpecimenRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+
+@Component
+public class FakeBookRepository implements BookRepository {
+    private Set<Book> bookSet = new HashSet<>();
+
+    @Autowired
+    SpecimenRepository specimenRepository;
+
+    @Override
+    public List<Book> getBooks() {
+        return new LinkedList<>(bookSet);
+    }
+
+    @Override
+    public Book getBookById(int id) {
+        return bookSet.stream()
+                .filter(b -> b.getId() == id)
+                .findAny()
+                .orElse(null);
+    }
+
+    @Override
+    public Book getBookByName(String name) {
+        return bookSet.stream()
+                .filter(b -> b.getName() == name)
+                .findAny()
+                .orElse(null);
+    }
+
+    public boolean setContainsName(Set<Book> set, String name) {
+        boolean flag = false;
+        for(Book book: set)
+        {
+            if(book.getName().equals(name))
+                flag = true;
+        }
+        return flag;
+    }
+
+    @Override
+    public void addToList(Book book) {
+        bookSet.add(book);
+    }
+
+    @Override
+    public void deleteSpecimenById(int id) {
+        bookSet.remove(getBookById(id));
+    }
+
+    @Override
+    public void updateUniqueBooks() {
```

Comment:
czy to jet funkcja, która po dodaniu egzemplarza dodaje automatycznie książkę?
bo jeżeli tak to skąd wiesz czy ona działa. 
Dajmy na to, że masz już 5 pozycji w bookSet, ale dodajesz nową. w 64 linijce inicjalizujesz pusty zbior, w 70 dodajesz do tego zbioru nowa ksiazke, ale pozniej zapominasz o tych 5 pozycjach, które już stworzyłeś.

do tego trzeba testy xd

Reply:
Nie wiem czy zwróciłeś uwagę ale tam jest for który leci po wszystkich dodanych egzemplarzach. Tak ta funkcja wymaga poprawy. podeślę drugą wersję